code/controllers/subsystem/ticker.dm

```diff
@@ -191,36 +194,73 @@ SUBSYSTEM_DEF(ticker)
 				send_tip_of_the_round()
 				tipped = TRUE
 
+			if(timeLeft <= 300 && !pre_setup_completed)
+				//Setup gamemode maps 30 seconds before roundstart.
+				if(!pre_setup())
+					//The mode failed to pre-setup
+					//This means that either there are insufficient players to run the gamemode
+					//Or there are no runnable gamemodes.
+					//In this case, we will resort to running extended.
+					to_chat(world, "<span class='boldannounce'>Pre-setup failed; Forcing gamemode to extended. Please file a github report including the current round ID: [GLOB.round_id].</span>")
+					message_admins("Error: [mode.setup_error]")
+					mode = null
+					GLOB.master_mode = "extended"
+					//Lets try this again.
+					if(!pre_setup())
```

Comment:
This isn't very elegant solution to the issue, especially since pre_setup is called again in GAME_STATE_SETTING_UP

Reply:
No its not because pre_setup_completed gets set to true

Reply:
It gets run again in `GAME_STATE_SETTING_UP`